tree-sitter-stack-graphs/src/cli/init.rs

```diff
@@ -103,154 +94,175 @@ impl InitArgs {
+            &self.language_file_extension
+        };
+        self.language_file_extension = Input::new()
             .with_prompt("Language file extension")
-            .validate_with(regex_validator(&VALID_NAME))
+            .with_initial_text(default_language_file_extension)
+            .validate_with(regex_validator(&VALID_CRATE_NAME))
             .interact_text()?;
-        println!();
 
         printdoc! {r#"
 
-            Give the NPM package name for this project. Must be a valid scoped or unscoped
-            package name.
+            Give the crate name for this project. May only contain letters, numbers, dashes,
+            and underscores.
             "#
         };
-        let project_npm_name: String = Input::new()
-            .with_prompt("Project NPM package name")
-            .with_initial_text("tree-sitter-stack-graphs-".to_string() + &language_id)
-            .validate_with(regex_validator(&VALID_NPM_PKG))
+        let default_crate_name = "tree-sitter-stack-graphs-".to_string() + &self.language_id;
+        self.crate_name = Input::new()
+            .with_prompt("Package name")
+            .with_initial_text(if self.crate_name.is_empty() {
+                &default_crate_name
+            } else {
+                &self.crate_name
+            })
+            .validate_with(regex_validator(&VALID_CRATE_NAME))
             .interact_text()?;
-        println!();
 
         printdoc! {r#"
 
-            Give the NPM package version for this project.
+            Give the crate version for this project. Must be in MAJOR.MINOR.PATCH format.
             "#
         };
-        let project_npm_version: String = Input::new()
-            .with_prompt("Project NPM package version")
-            .with_initial_text("0.1.0")
-            .validate_with(regex_validator(&VALID_NPM_VERSION))
+        self.crate_version = Input::new()
+            .with_prompt("Package version")
+            .with_initial_text(if self.crate_version.is_empty() {
+                "0.1.0"
+            } else {
+                &self.crate_version
+            })
+            .validate_with(regex_validator(&VALID_CRATE_VERSION))
             .interact_text()?;
-        println!();
 
         printdoc! {r#"
 
             Give the project author in the format NAME <EMAIL>. Leave empty to omit.
             "#
         };
-        let project_author: String = Input::new()
+        self.author = Input::new()
             .with_prompt("Author")
+            .with_initial_text(&self.author)
             .allow_empty(true)
             .interact_text()?;
-        println!();
 
         printdoc! {r#"
 
             Give the project license as an SPDX expression. Leave empty to omit.
             "#
         };
-        let project_license: String = Input::new()
+        self.license = Input::new()
             .with_prompt("License")
+            .with_initial_text(&self.license)
             .allow_empty(true)
             .interact_text()?;
```

Comment:
Optional suggestion:  having a selection of common licenses in the correct format to choose from might be a nice convenience.

Not entirely sure this would be the right thing but something like:

```rust
let license = Select::new()
    .items(&["APACHE-2.0", "MIT", "Other"])
    .default(0)
    .interact()?;
```

And if the selection is 2 (i.e. "Other"),  then use the existing logic.

Reply:
I like this suggestion. Implementing it got a bit out of hand 🙈. See 593ebe3.

Reply:
👍 this is amazing!  I love it even includes the license text for you ❤️